PennMobile/src/main/java/com/pennapps/labs/pennmobile/LaundryMachineFragment.java

```diff
@@ -119,9 +135,47 @@ public void onResume() {
 
     @Override
     public void onDestroyView() {
-        super.onDestroyView();
-        mActivity.setTitle(R.string.laundry);
+
+        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(getActivity());
+        SharedPreferences.Editor editor = sp.edit();
+        editor.putBoolean(laundryRoom.name + "_isFavorite", favoriteState);
+        if(laundryRoom.name.contains("-")) {
+            editor.putBoolean(laundryRoom.name.substring(0, laundryRoom.name.indexOf("-")) + "_isFavorite", favoriteState);
+        }
+        editor.commit();
+
         mActivity.removeTabs();
+        mActivity.setTitle("Laundry");
+        super.onDestroyView();
         ButterKnife.unbind(this);
```

Comment:
This is a bad design.  You have to save the details after they are clicked, not when they are destroyed.  This is because "onDestroyView" is killable and is not guaranteed to be run.